File: .claude/hooks/protect-paths.mjs
#!/usr/bin/env node
// @hookstack pre-edit-protect-paths
// Protège les fichiers sensibles contre l'écriture (PreToolUse Write|Edit)
import { readFileSync } from 'node:fs';
import { fileURLToPath } from 'node:url';

const PROTECTED = [
  /\/\.env$/,
  /\/\.env\.local$/,
  /\/\.env\.production/,
  /\/secrets\//,
  /\/(id_rsa|id_ed25519|.*\.pem)$/,
];

export function run(input) {
  const filePath = input.tool_input?.file_path ?? '';
  const blocked = PROTECTED.find((p) => p.test(filePath));
  return blocked
    ? { decision: 'block', reason: `Fichier protégé : ${filePath}. Modifiez manuellement si intentionnel.` }
    : null;
}

/* v8 ignore next 5 */
if (process.argv[1] === fileURLToPath(import.meta.url)) {
  const input = JSON.parse(readFileSync(0, 'utf8'));
  const result = run(input);
  if (result) process.stdout.write(JSON.stringify(result));
}

File: .claude/hooks/protect-paths.py
#!/usr/bin/env python3
# @hookstack pre-edit-protect-paths
"""Protects sensitive files from being written (PreToolUse Write|Edit)."""
import json
import re
import sys

PROTECTED = [
    re.compile(r"/\.env$"),
    re.compile(r"/\.env\.local$"),
    re.compile(r"/\.env\.production"),
    re.compile(r"/secrets/"),
    re.compile(r"/(id_rsa|id_ed25519|.*\.pem)$"),
]


def run(input_data):
    file_path = (input_data.get("tool_input") or {}).get("file_path") or ""
    for pattern in PROTECTED:
        if pattern.search(file_path):
            return {
                "decision": "block",
                "reason": f"Fichier protégé : {file_path}. Modifiez manuellement si intentionnel.",
            }
    return None


if __name__ == "__main__":
    data = json.load(sys.stdin)
    result = run(data)
    if result:
        sys.stdout.write(json.dumps(result))
